metadata-ingestion/src/datahub/ingestion/graph/client.py

```diff
@@ -717,6 +718,23 @@ def soft_delete_urn(
             )
         )
 
+    def exists(self, entity_urn: str) -> bool:
+        entity_urn_parsed: Urn = Urn.create_from_string(entity_urn)
+        try:
+            key_aspect_class = KEY_ASPECTS.get(entity_urn_parsed.get_type())
+            if key_aspect_class:
+                result = self.get_aspect(entity_urn, key_aspect_class)
+                return result is not None
+            else:
+                raise Exception(
+                    f"Failed to find key class for entity type {entity_urn_parsed.get_type()} for urn {entity_urn}"
+                )
+        except Exception as e:
```

Comment:
this exception handling might be too broad - if we get a connection error, we really should be raising an exception and not swallowing it and returning false